src/lsm/compaction.zig

```diff
@@ -965,19 +981,16 @@ pub fn CompactionType(
                 const value_block_checksums = index_schema.data_checksums_used(index_block);
 
                 var free_blocks_used: usize = 0;
+                // Read at most half of the blocks, since the other half are reserved for the
```

Comment:
This isn't the case here; all of `source_value_blocks[...]` are reserved for input blocks, but we want to smooth out the reading over multiple steps to reduce latency spikes.

Reply:
Updated to

```
                // Read at most half of the blocks, to smooth out reading over multiple steps.
```